src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1452,4 +1460,76 @@ export class NetworkAreaDiagramViewer {
             }
         }
     }
+
+    public setJsonBranchStates(branchStates: string) {
+        const branchStatesArray: BranchState[] = JSON.parse(branchStates);
+        this.setBranchStates(branchStatesArray);
+    }
+
+    public setBranchStates(branchStates: BranchState[]) {
+        branchStates.forEach((branchState) => {
+            if (!this.edgesMap.has(branchState.branchId)) {
+                const edge: EdgeMetadata | undefined = this.diagramMetadata?.edges.find(
```

Comment:
I get an error on find on my IDE "Argument types do not match parameters", because you cannot call find on undefined with the given arguments. This can be fixed like this:
```suggestion
                const edge: EdgeMetadata | undefined = (this.diagramMetadata?.edges ?? []).find(
```